sanic/mixins/routes.py

```diff
@@ -702,6 +712,8 @@ def static(
         content_type: Optional[bool] = None,
         apply: bool = True,
         resource_type: Optional[str] = None,
+        autoindex: bool = False,
+        index_name: Union[str, Default] = _default,
```

Comment:
I've been thinking of how those arguments could be better. I don't associate autoindex with file browser but I see that Apache and Nginx have such named modules, where Apache also includes index.html while Nginx apparently does not.

If we take the two features being closely related, perhaps overload a single argument (autoindex) for both? This also makes it easier to pass it around (I see that the two appear in five methods' arguments in this PR).

```python
autoindex=False                 # Disable both
autoindex=True                  # Enable file browser only
autoindex="index.html"          # Only index.html, no file browser
autoindex=(True, "index.html")  # Enable both, be explicit on index.html
```

The rule then would be to take a list/tuple of arguments by type:
* bool for file browser
* str for filename(s) to look for

For convenience allow leaving out the tuple when there is only one argument. Being explicit on index.html handling with the file manager avoids accidentally enabling them where only file listings are intended (like with the nginx module). This is more lightweight than the `ssl=` argument of `app.run` that is similarly overloaded for many things and such a list should be more than sufficient for this (using a dict instead seems overkill, bad for syntax).

Perhaps you have other idea of simplification, but the current two separate arguments and how they function are itching me in a way that says that it probably isn't quite right.

Reply:
Interesting. I do get the concept, and think that `autoindex` is the right term since (as you pointed out) we are not inventing it. I also am not entirely happy with the `index_name` argument. It does certainly need to be an implementation detail. Such overloading may be confusing, but if we type annotate it properly with `@overload` and put sufficient docs in for it, maybe not.

Reply:
The problem that I have is that:

```py
autoindex=("index.html", True)
```

That is really what we want to "normal" usecase to be when implementing this. File browser should not be enabled, except when explicit. And, enabling both the browser and `index.html` fallback is a bit cumbersome.

Reply:
Hmmm.... I don't know. This feels like a complication. Nginx is nice and explicit on both arguments: [autoindex](https://nginx.org/en/docs/http/ngx_http_autoindex_module.html) and [index](https://nginx.org/en/docs/http/ngx_http_index_module.html#index).

Reply:
What about making it two args, but you only need two in certain circumstances

```python
autoindex=False                 # Disable both, DEFAULT
autoindex=True                  # Enable both
autoindex="index.html"          # Enable both, more explicit form of previous
autoindex=("index.html", "index.txt")
                                # Try in order (browser still enabled as last resort)
autoindex=True, browser=False   # Index only, explicit disable browser
```

Reply:
I don't like enabling file browser "by default". `--simple` handles that use case naturally (enabling both), but I think it would rarely be needed in webapps.

Considering modern webapps, that serve a static front (something generated by React/VueJS/etc, or written by hand) at site root and only make API calls to Sanic, the typical use would look like:

```python
app = Sanic(__name__)
app.static("/", staticdir, autoindex="index.html")

@app.websocket("/ws")  # or some JSON methods, or a Blueprint for the whole API
...
```

This webapp should not show any file listings even if the user happens to find a static directory URL other than the root (assuming there is index.html only at root).

Being explicit about both the file browser `True` and the index filename(s) keeps things secure and unsurprising. If both are needed, they are still easily enabled manually (that's why a tuple rather than dict or two kwargs). Agreed, giving the name "index.html" is a bit superfluous because it is always called that, but again it is self-documenting and has a clear meaning.

Reply:
Explicit APIs that are self-documenting are better than implicit. The tuple method can lead to confusion, and/or feel somewhat "magical" in that it is not clear on the surface what it is doing.

What if we just rename the arguments?
- `autoindex => browser` - Whether to fallback to browser
- `index_name => index` - Single or list to be checked in order

Reply:
That works for me. But I do prefer only keeping `autoindex`, for the reasons in my previous message and in the initial proposal (first message in this thread).